grpc-api/src/main/java/io/deephaven/grpc_api/console/ConsoleServiceGrpcImpl.java

```diff
@@ -81,6 +90,18 @@ public ConsoleServiceGrpcImpl(final Map<String, Provider<ScriptSession>> scriptT
         if (!scriptTypes.containsKey(WORKER_CONSOLE_TYPE)) {
             throw new IllegalArgumentException("Console type not found: " + WORKER_CONSOLE_TYPE);
         }
+
+        ProcessEnvironment.getGlobalFatalErrorReporter().addInterceptor((message, throwable) -> {
+            final long now = scheduler.currentTime().getNanos();
+            final LogBufferRecord record = new LogBufferRecord();
+            record.setLevel(LogLevel.ERROR);
+            record.setTimestampMicros(now);
+
+            record.setData(ByteBuffer.wrap(message.getBytes(StandardCharsets.UTF_8)));
+            logBuffer.record(record);
+            record.setData(ByteBuffer.wrap(throwable.getMessage().getBytes(StandardCharsets.UTF_8)));
```

Comment:
This pattern seems wrong. Records aren't re-usable in this way, because you'll overwrite the one in the log buffer's circular array.

Reply:
It's "safe" for your listener, but not for `LogBuffer` itself.